File: Leet_Code-128/Leet_Code-128.py
'''
Question 128
'''

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums = list(set(nums))
        nums.sort()
        consts = [1 if len(nums)>=1 else 0]
        cons = 1
        for i in range(1,len(nums)):
            if nums[i] == nums[i-1] + 1:
                cons += 1
            else:
                consts.append(cons)
                cons = 1
            consts.append(cons)
        return max(consts)

File: Leet_Code-128/Leet_Code-128.java
/*
 * Question 128
 */

import java.util.*;
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set_nums = new HashSet<>();
        for (int num : nums) {
            set_nums.add(num);
        }
        ArrayList<Integer> nums_ord = new ArrayList<>(set_nums);
        Collections.sort(nums_ord);
        ArrayList<Integer> consts = new ArrayList<>();
        if (nums_ord.size() >= 1){
            consts.add(1);
        } else {
            consts.add(0);
        }
        int cons = 1;

        for(int i = 1; i < nums_ord.size(); i++) {
            if (nums_ord.get(i) == nums_ord.get(i-1) + 1) {
                cons += 1;
            } else{
                consts.add(cons);
                cons = 1;
            }
            if (i == nums_ord.size()-1){
                consts.add(cons);
            }
        }
        return Collections.max(consts);
    }
}